pkg/util/finalizers/finalizers.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package finalizers
+
+// Remove returns a newly created []string that contains all items from slice that
+// are not equal to s and modifier(s) in case modifier func is provided.
+func Remove(finalizers []string, toRemove string) []string {
+	newSlice := make([]string, 0, len(finalizers))
+	for _, item := range finalizers {
+		if item == toRemove {
+			continue
+		}
+		newSlice = append(newSlice, item)
+	}
+	if len(newSlice) == 0 {
+		// Sanitize for unit tests so we don't need to distinguish empty array
+		// and nil.
+		newSlice = nil
+	}
+	return newSlice
+}
```

Comment:
Would it make sense to use [slice.DeleteFunc](https://pkg.go.dev/slices#DeleteFunc) instead?

Reply:
I would have to adapt the tests that check for nil 🤔, unless you are okay with just changing the contents of the function with the `slice.DeleteFunc`